Core/Core/Planner/CalendarToDo/ViewModel/CalendarToDoDetailsViewModel.swift

```
@@ -17,17 +17,40 @@
 //
 
 import Foundation
+import Combine
 
 public class CalendarToDoDetailsViewModel: ObservableObject {
     public let navigationTitle = String(localized: "To Do", bundle: .core)
-    public let title: String?
-    public let date: String?
-    public let description: String?
     public let screenConfig = InstUI.BaseScreenConfig(refreshable: false)
 
-    public init(plannable: Plannable) {
-        title = plannable.title
-        date = plannable.date?.dateTimeString
-        description = plannable.details
+    @Published public private(set) var title: String?
+    @Published public private(set) var date: String?
+    @Published public private(set) var description: String?
+    @Published public private(set) var navBarColor: UIColor?
+
+    private let plannable: Plannable
+    private var subscriptions = Set<AnyCancellable>()
+
+    public init(plannable: Plannable, interactor: CalendarToDoInteractor) {
+        self.plannable = plannable
+
+        interactor.getToDo(id: plannable.id)
+            .sink { [weak self] in
+                self?.title = $0.title
+                self?.date = $0.date?.dateTimeString
+                self?.description = $0.details
+                self?.navBarColor = $0.color.ensureContrast(against: .backgroundLightest)
+            }
+            .store(in: &subscriptions)
```

I see that this function fetches data from the db only. How come we swallow every error? If something fails during the Core Data fetch we'll have an empty white screen.

---

The assumption was that this screen is opened directly from a list which already contains the `plannable` and this update is there only for automatic updates after editing.
But you are right, it should be handled to make it more robust.
Updated to fallback to the input `plannable` in case of any error.